portfolio/src/main/java/com/google/sps/servlets/DeleterServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that will delete all comments. */
+@WebServlet("/delete-comments")
+public class DeleterServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
Do you have to query all of them and then delete them one by one? I thought datastore will have APIs for delete all.

If you have to query all of them first, I think you don't need to addSort since sort doesn't matter here and adding sort might make the query less efficient.

Reply:
found a datastore tool (in fetchoptions) to do this

Reply:
I could not find an API for deleteAll. What I found was this:
"If you need to delete entities in bulk, we recommend using Dataflow, a managed service for developing and executing data processing workflows. Dataflow provides a Datastore delete template that you can use to delete entities selected by a GQL query."
This is a paid feature it looks like.

Removed the sort.